app/controllers/CheckDetailsController.scala

```diff
@@ -66,23 +64,23 @@ class CheckDetailsController @Inject()(
       }
     }
 
+
   private def pdvCheck(mode: Mode,
                        origin: Option[String])
-                      (implicit hc: HeaderCarrier, request: DataRequest[AnyContent]): Future[Result] = {
+                      (implicit hc: HeaderCarrier, request: DataRequestWithUserAnswers[AnyContent]): Future[Result] = {
     val pdvRequest = PDVRequest(
       request.credId.getOrElse(EmptyString),
-      request.session.data.getOrElse("sessionId", EmptyString)
+      hc.sessionId.map(_.value).getOrElse(EmptyString)
     )
     personalDetailsValidationService.getPDVData(pdvRequest) flatMap {
```

Comment:
Looks like this PDV call is being actioned twice on this controller, one in action class and once here, where it also stores the data into a Mongo DB. I wonder if we need to store in Mongo at all though? Might it not be better to just call PDV each time to get the data (on the action class)?

Reply:
Yes in this specific instance it is called twice. Because for the first time you got to call the method getPDVData and for rest of the subsequent calls you need getPDVDataMatching. I could have created one more action class to call that method only to be used in this controller. But I felt it would be not worth saving one extra call. About calling calling pdv directly without storing, would increase few calls to the backend. But yes we can do it, but I don't see any benefit doing that.